imagehash/__init__.py

```diff
@@ -116,10 +116,10 @@ def average_hash(image, hash_size=8):
 def phash(image, hash_size=32):
 	image = image.convert("L").resize((hash_size, hash_size), Image.ANTIALIAS)
 	pixels = numpy.array(image.getdata(), dtype=numpy.float).reshape((hash_size, hash_size))
-	dct = scipy.fftpack.dct(pixels)
-	dctlowfreq = dct[:8, 1:9]
-	avg = dctlowfreq.mean()
-	diff = dctlowfreq > avg
+	dct = scipy.fftpack.dct(scipy.fftpack.dct(pixels, axis=0), axis=1)
+	dctlowfreq = dct[:8, :8]
```

Comment:
I have to disagree -- there is neither a row of constants nor a column of constants.  Only the very first coefficient, [0, 0] is a constant, a zero-power coefficient.  The remaining entries in the top row and left column are first-power low-frequency coefficients in each axis.  This is why I keep saying we should use [:8, :8] (or / hash_size, whatever you all want) instead of trying to offset by a row or column.  Please see my issue report for more details:  https://github.com/JohannesBuchner/imagehash/issues/13

Another thing to keep in mind: it's unlikely that any two implementations of a fuzzy image hash will ever return the same hash value as each other.  A lot of differences stack up:  do we convert to greyscale and then resize down, or resize down and then convert to greyscale; what antialiasing algorithm is used; what endian-ness is used when converting the array of coefficients to bits in an integer; and probably also issues of simple numerical precision and handling in the low-level libraries.